addons/udes_stock/models/stock_move_line.py

```diff
@@ -252,3 +252,139 @@ def get_move_lines_ordered_by(self, domain, aux_domain=None, order="id"):
         if aux_domain is not None:
             domain += aux_domain
         return StockMoveLine.search(domain, order=order)
+
+    def _find_move_lines(self, uom_qty, product=None, package=None, lot_name=None, location=None):
+        """ Find a subset of move lines from self matching the subset of the key:
+            product, package, lot and location.
+        """
+        funcs = []
+        if product:
+            funcs.append(lambda ml: ml.product_id == product)
+        if package:
+            funcs.append(lambda ml: ml.package_id == package)
+        if location:
+            funcs.append(lambda ml: ml.location_id == location)
+        if lot_name:
+            funcs.append(lambda ml: (lot_name == lot_name or ml.lot_id.name == lot_name))
+        func = lambda ml: all(f(ml) for f in funcs)
+
+        # Find move lines
+        mls = self.filtered(func)
+        mls_fulfill, new_ml, uom_qty = mls.move_lines_for_qty(uom_qty)
+        if uom_qty > 0:
+            # Note: when implementing add_unexpected_parts() review if this is the best place
+            mls_fulfill |= self.picking_id.add_unexpected_parts()
+        return mls_fulfill, new_ml
+
+    def prepare_move_lines(
```

Comment:
Bit of a nit, but if `self` is an instance of `StockMoveLine` we could just call this method "prepare" - likewise in other method names that duplicate the class name (assuming that they aren't inherited).